scripts/gha_unit_tests_collector.py

```diff
@@ -13,29 +13,27 @@ def get_all_unit_test_shards():
     return unit_test_workflow_shards
 
 
-def get_modules_except_cms():
+def get_unit_test_modules(module_name="lms"):
+    unit_test_modules = set()
     all_unit_test_shards = get_all_unit_test_shards()
-    return [paths for shard_name, paths in all_unit_test_shards.items() if not paths.startswith('cms')]
-
-
-def get_cms_modules():
-    all_unit_test_shards = get_all_unit_test_shards()
-    return [paths for shard_name, paths in all_unit_test_shards.items() if paths.startswith('cms')]
+    for shard_name, shard_confing in all_unit_test_shards.items():
+        if shard_confing.get('path')[0].startswith('cms') and module_name == "cms":
```

Comment:
This would be easier to understand if you do "shard_confing.get('path')[0].startswith('cms')" once and put it in a variable.  Or:
```
if shard_confing.get('path')[0].startswith('cms'):
    if module_name == "cms":
        unit_test_modules.update(shard_confing["path"])
else:
    if module_name != "cms":
        unit_test_modules.update(shard_confing["path"])
```
Note since shard_config should always have "path", then `["path"]` is better than `.get("path")`.